tests/Microsoft.Identity.Test.E2e.netcore/ManagedIdentityAzureArcTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Test.E2E
+{
+    [TestClass]
+    public class ManagedIdentityAzureArcTests
+    {
+        private const string ArmScope = "https://management.azure.com";
+
+        /// <summary>
+        /// Check if the test is running on Azure Arc.
+        /// </summary>
+        /// <returns></returns>
+        private static bool IsArc() =>
+            !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("IDENTITY_ENDPOINT"));
+
+        /// <summary>
+        /// Builds a System-Assigned MI app with *per-instance* in-memory cache
+        /// to avoid cross-test pollution.
+        /// </summary>
+        private static IManagedIdentityApplication BuildSami()
```

Comment:
nit: you are adding too many comments. The method name is indicative of what it does. Avoid adding comments explaining the "WHAT", add comments explaining the "WHY" (not needed here) i.e. surprising behavior

Reply:
removed